### PR TITLE
12996 ad aux dot fix

### LINKED ISSUE #12996: AD coupled time with aux fails
## Rationale
Supplying an aux var to the getter `adCoupledDot` causes issues where the AD time deriv is very large instead of zero. Similar to #12921 but effects `adCoupledDot` instead of `adCoupledValue`?

## Description

https://github.com/veeshy/moose/tree/ad_coupled_time_test adds a test kernel, `ADCoupledTimeTest` that takes a coupled var in. `test/tests/kernels/ad_coupled_value/ad_aux_coupled_time_value.i` gives an aux coupled var and the solve does not converge. The v_dot term added should be zero too. If you input a constant number like `v = 2` instead of `v = v` the problem works fine.

I found on OSX 10.14 this will mostly pass and randomly fail, on Ubuntu 18.04 it will always fail.

## Impact

Allow for coupled aux vars in AD time kernel.

### PR DESCRIPTION
**Pull request by veeshy**
Coupled aux variables in an AD time kernel was crashing, possibly because the AD system doesn't support aux variable derivatives but it tried anyways. 

I made a test that fails w/o the change in `MooseVariableFE.C`, I'm not sure if I hit all the relevant bits in `MooseVariableFE.C` though. Test passes with those changes so it seems like a step in the right direction.

Closes #12996.

### CODE REVIEW

**Comment by veeshy** [comment]
@lindsayad I took a stab at this, the check to not do any AD aux stuff might be late, wasn't sure the best place to put it so I did it when the computed ad dot value gets assigned.

File: test/src/kernels/ADCoupledTimeTest.C [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ADCoupledTimeTest<compute_stage>::ADCoupledTimeTest(const InputParameters & parameters)
+  : ADTimeKernel<compute_stage>(parameters), _v_dot(adCoupledDot("v"))
+{
+}
+
+template <ComputeStage compute_stage>
+ADResidual
+ADCoupledTimeTest<compute_stage>::precomputeQpResidual()
+{
+  return _test[_i][_qp] * _v_dot[_qp];
```
Comment (lindsayad) [inline test/src/kernels/ADCoupledTimeTest.C]: This function will get multiplied internally by `_test` in order to save unnecessary operations (generally speaking; in this particular single operation case there is no savings) so you need to remove `_test`. This is the reason for the CIVET failures.

File: framework/src/variables/MooseVariableFE.C [hunk 2]
```diff
@@ -1409,7 +1409,7 @@ MooseVariableFE<OutputType>::computeAD(
           _ad_second_u[qp] += _ad_dof_values[i] * (*second_phi)[i][qp];
       }
 
-      if (is_transient)
+      if (is_transient && _var_kind == Moose::VAR_NONLINEAR)
```
Comment (lindsayad) [inline framework/src/variables/MooseVariableFE.C]: I don't think is the right fix per se. There must be something stupid happening in the `TimeIntegrator::computeADTimeDerivatives` method which I'm going to look into now

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Comment by lindsayad** [comment]
See lindsayad/moose commits 68416a1cc1..6f05b. I did some rebasing and autosquashing so most of my changes are included in your "Added test to show a potential fix" commit. If they look good to you, then I will force push to your branch.

**Comment by lindsayad** [comment]
With the changes I've introduced we should actually be able to get a non-zero time derivative for an aux variable retrieved through `adCoupledDot`, e.g. if the aux variable is changing through the time, then we should be able to get it's correct time derivative. Would you want to update your test to show this?

**Comment by veeshy** [comment]
Glad you found a smarter spot for the fix! I'll update the test to include a time varying aux after you force push.

**Comment by lindsayad** [comment]
Cool. Just pushed

**Comment by veeshy** [comment]
Oops, I think I lost the point here, I still need to make a test that couples an *aux* var varying

**Comment by veeshy** [comment]
Okay, test added!

**Comment by lindsayad** [comment]
Looks like a different mesh was used for the gold file and the current test...

**Comment by veeshy** [comment]
yup -_-, added the new gold

**Comment by lindsayad** [comment]
If you're going to use FDP as a preconditioner, you'll need to specify `allow_warnings = true` in your test block

**Comment by veeshy** [comment]
Hm, no reason fdp is needed, changed it to smp

**Review by lindsayad — APPROVED** [review]

**Comment by lindsayad** [comment]
@veeshy Thanks for the contribution!

**Pull request merged**